File: Hoseol/Programmers/정렬/k번째수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;


vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
   int copy[101];
    int i;int j; int k;
    for(int a=0;a<commands.size();a++)
    {
        for(int b=0;b<array.size();b++)
        {
            copy[b]=array[b];
        }
        
          i=commands[a][0];
          j=commands[a][1];
          k=commands[a][2]-2+i;
        sort(copy+i-1,copy+j);
      
            answer.push_back(copy[k]);
        
       
        
        
    }

    
    

    return answer;
}

File: Jeongmi/Programmers/정렬/K번째수.java
package practice.programmers.정렬;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class K번째수 {
    private static final Scanner SCANNER = new Scanner(System.in);
    private static final String DELIMITER = ", ";

    public static void main(String[] args) {
        int[] array = {1, 5, 2, 6, 3, 7, 4};
        int[][] commands = {{2, 5, 3}, {4, 4, 1}, {1, 7, 3}};
        solution(array, commands);
    }

    private static int[] solution(int[] array, int[][] commands) {
        List<Integer> answer = new ArrayList<>();

        for (int[] command : commands) {
            int[] partOfArray = Arrays.copyOfRange(array, command[0] - 1, command[1]);
            Arrays.sort(partOfArray);
            answer.add(partOfArray[command[2] - 1]);
        }

        System.out.println(answer);
        return convertIntegers(answer);
    }

    private static int[] convertIntegers(List<Integer> integers) {
        int[] result = new int[integers.size()];
        for (int i = 0; i < result.length; i++) {
            result[i] = integers.get(i);
        }
        return result;
    }
}
